go/libraries/doltcore/merge/merge.go

```diff
@@ -51,99 +51,101 @@ func NewMerger(ctx context.Context, root, mergeRoot, ancRoot *doltdb.RootValue,
 // MergeTable merges schema and table data for the table tblName.
 func (merger *Merger) MergeTable(ctx context.Context, tblName string, sess *editor.TableEditSession) (*doltdb.Table, *MergeStats, error) {
 	tbl, ok, err := merger.root.GetTable(ctx, tblName)
-
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var h hash.Hash
 	if ok {
 		h, err = tbl.HashOf()
-
 		if err != nil {
 			return nil, nil, err
 		}
 	}
 
 	mergeTbl, mergeOk, err := merger.mergeRoot.GetTable(ctx, tblName)
-
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var mh hash.Hash
 	if mergeOk {
 		mh, err = mergeTbl.HashOf()
-
 		if err != nil {
 			return nil, nil, err
 		}
 	}
 
 	ancTbl, ancOk, err := merger.ancRoot.GetTable(ctx, tblName)
-
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var anch hash.Hash
+	var ancTblSchema schema.Schema
 	if ancOk {
 		anch, err = ancTbl.HashOf()
-
+		if err != nil {
+			return nil, nil, err
+		}
+		ancTblSchema, err = ancTbl.GetSchema(ctx)
 		if err != nil {
 			return nil, nil, err
 		}
 	}
 
-	if ok && mergeOk && h == mh {
-		return tbl, &MergeStats{Operation: TableUnmodified}, nil
-	}
-
-	if !ancOk {
-		if mergeOk && ok {
-			return nil, nil, ErrSameTblAddedTwice
-		} else if ok {
-			return tbl, &MergeStats{Operation: TableUnmodified}, nil
+	{ // short-circuit logic
+		if ancOk && schema.IsKeyless(ancTblSchema) {
+			if ok && mergeOk && ancOk && h == mh && h == anch {
```

Comment:
Would give this logic a method and a name to give a clue to the intent behind the logic